src/process.ts

```diff
@@ -0,0 +1,65 @@
+import * as githubActions from "@actions/github";
+
+import { findReferendum } from "./find-referendum";
+import { parseRFC } from "./parse-RFC";
+import { RequestResult, RequestState } from "./types";
+
+const blockHashInstructions =
+  `<details><summary>Instructions to find the block hash</summary>` +
+  "Here is one way to find the corresponding block hash." +
+  `\n\n1. Open the referendum on Subsquare.` +
+  "\n\n2. Switch to the `Timeline` tab." +
+  `\n<img width="250px" src="https://raw.githubusercontent.com/paritytech/rfc-propose/main/src/images/timeline_tab.png" />` +
+  "\n\n---" +
+  "\n\n3. Go to the details of the `Confirmed` event." +
+  `\n<img width="480px" src="https://raw.githubusercontent.com/paritytech/rfc-propose/main/src/images/confirmed_event.png" />` +
+  "\n\n---" +
+  `\n\n2. Go to the details of the block containing that event.` +
+  `\n<img width="380px" src="https://raw.githubusercontent.com/paritytech/rfc-propose/main/src/images/block_number.png" />` +
+  "\n\n---" +
+  `\n\n2. Here you can find the block hash.` +
+  `\n<img width="620px" src="https://raw.githubusercontent.com/paritytech/rfc-propose/main/src/images/block_hash.png" />` +
+  `\n</details>`;
+
+export const handleProcessCommand = async (
+  requestState: RequestState,
+  blockHash: string | undefined,
+): Promise<RequestResult> => {
+  const parseRFCResult = await parseRFC(requestState);
+  if ("success" in parseRFCResult) {
+    return parseRFCResult;
+  }
+  if (!blockHash && !blockHash?.startsWith("0x")) {
+    return {
+      success: false,
+      errorMessage:
+        "Please provider a block hash where the referendum confirmation event is to be found.\n" +
+        "For example:\n\n" +
+        "```\n/rfc process 0x39fbc57d047c71f553aa42824599a7686aea5c9aab4111f6b836d35d3d058162\n```\n\n" +
+        blockHashInstructions,
+    };
+  }
+
+  const referendum = await findReferendum({ parseRFCResult, blockHash });
+  if (!referendum) {
+    return {
+      success: false,
+      errorMessage: `Unable to find the referendum confirm event in the given block.\n\n` + blockHashInstructions,
+    };
+  }
+  if ("approved" in referendum && referendum.approved) {
```

Comment:
if we've created an RFC and the hash of .md file is X, but then until it's approved and processed, the hash has been changed (someone pushed some update to file) - then we should check if before merge? and either propose to create a new referendum with a new file hash or roll back to an approved commit, right?

Reply:
If some pushes some update and the hash changes, then the referendum will not be found by the action.
It will respond back with a "Unable to find the referendum" message and the PR will not be merged.

Reply:
ah that makes sense!
if someone pushes the updates to file during ongoing referendum should we do something with it or suggest some actions?

Reply:
Hmm, we would have to check every commit on the PR branch, and if we find one that matches the ongoing situation that means we detected this situation and can suggest actions (e.g. suggest to roll-back to that commit).

I have created a separate issue for this: https://github.com/paritytech/rfc-action/issues/11